### PR TITLE
psst: init at 0.1.0

### PR DESCRIPTION
**Pull request by vbrandl**
###### Motivation for this change

psst is an alternative Spotify client 

https://github.com/jpochyla/psst

Solves https://github.com/NixOS/nixpkgs/issues/115155

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vbrandl** [comment]
I just realized, this is a duplicate of https://github.com/NixOS/nixpkgs/pull/134921

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -13426,4 +13426,10 @@
     github = "jpagex";
     githubId = 635768;
   };
+  vbrandl = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:13429]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -28523,6 +28523,8 @@ with pkgs;
 
   spotifywm = callPackage ../applications/audio/spotifywm { };
 
+  psst = callPackage ../applications/audio/psst/default.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:28526]:
```suggestion
  psst = callPackage ../applications/audio/psst { };
```

File: pkgs/applications/audio/psst/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchFromGitHub {
+    owner = "jpochyla";
+    repo = pname;
+    rev = "f6eb1fe8d273af43266407d998aacac87df8c04d";
+    sha256 = "0w939w7sdalkms63dzgq6s12hwxngbhbqy7rvnq4wyn10g8hv5y1";
+  };
+
+  cargoSha256 = "1zrdd82rdshhczlyndn0g847jq0f801sj27kggyky4jxzl89p28s";
+
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/psst/default.nix:17]: meta should be last.

File: pkgs/applications/audio/psst/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    description = "Fast and multi-platform Spotify client with native GUI";
+    homepage = "https://github.com/jpochyla/psst";
+    license = licenses.mit;
+    maintainers = [ maintainers.vbrandl ];
+  };
+
+  buildInputs = with pkgs; [
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/psst/default.nix:24]:
```suggestion
  buildInputs = [
```

File: pkgs/applications/audio/psst/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    atk
+    cairo
+    dbus
+    gdk-pixbuf
+    glib
+    gtk3
+    pango
+  ];
+
+  nativeBuildInputs = with pkgs; [
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/psst/default.nix:35]:
```suggestion
  nativeBuildInputs = [
```
native should come before buildInputs.

File: pkgs/applications/audio/psst/default.nix [hunk 6]
```diff
@@ -0,0 +1,38 @@
+{ pkgs, lib, fetchFromGitHub, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "psst";
+  version = "0.1.0";
+  buildAndTestSubdir = "psst-gui";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/psst/default.nix:6]: Is this cargo related?

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (vbrandl) [reply maintainers/maintainer-list.nix:13429]: Done in df432c38cfb

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (vbrandl) [reply pkgs/top-level/all-packages.nix:28526]: Done in 32b11de94ff

File: pkgs/applications/audio/psst/default.nix (hunk 3 above)
Comment (vbrandl) [reply pkgs/applications/audio/psst/default.nix]: Done in 32b11de94ff

File: pkgs/applications/audio/psst/default.nix (hunk 4 above)
Comment (vbrandl) [reply pkgs/applications/audio/psst/default.nix]: Done in 32b11de94ff

File: pkgs/applications/audio/psst/default.nix (hunk 6 above)
Comment (vbrandl) [reply pkgs/applications/audio/psst/default.nix:6]: Yes. The repository is a crago workespace with multiple crates. The `psst-gui` subdirectory contains the actual binary project. The `buildAndTestSubdir` tells cargo which which directory to build.

https://github.com/jpochyla/psst/tree/master/psst-gui

**Comment by vbrandl** [comment]
I tried fixing your comments, still https://github.com/NixOS/nixpkgs/pull/134921 looks like the more fleshed out PR since it also works on darwin

File: pkgs/applications/audio/psst/default.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/applications/audio/psst/default.nix:6]: Lets move it below cargoSha256 to make that clear.

Comment (vbrandl) [reply pkgs/applications/audio/psst/default.nix]: Done in 9adf5fd

File: pkgs/applications/audio/psst/default.nix (hunk 4 above)
Comment (vbrandl) [reply pkgs/applications/audio/psst/default.nix]: What's the reason, why you proposed this change? If I don't put `with pkgs;` before the list, I have to list all the inputs as `pkgs.pkg-config` and so on, when using `nix-build -A psst`. Is there any difference between the two or is it just best practice?

File: pkgs/applications/audio/psst/default.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
+{ pkgs, lib, fetchFromGitHub, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "psst";
+  version = "0.1.0";
```
[the remark this answers is not in the document]
Comment (vbrandl) [reply pkgs/applications/audio/psst/default.nix]: Done in eaef603fcf50d5d929cb43542e6b5744e0faa413.

Also merged the current master to fix the merge conflict.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Also please squash all psst related commits except the maintainer entry into one commit.

**Comment by vbrandl** [comment]
I squashed everything except the maintainer in 8547a003aab9ee74d88a746233bd25e9e1339288 and applied your change requests
